dropshot/src/handler.rs

```diff
@@ -139,6 +150,21 @@ impl<Context: ServerContext> RequestContext<Context> {
              */
             .unwrap_or(server_config.page_default_nitems))
     }
+
+    /**
+     * Add a header that will be included with the HTTP response.
+     */
+    pub fn add_response_header(
+        self: Arc<Self>,
+        header: http::header::HeaderName,
+        value: http::header::HeaderValue,
```

Comment:
What do we want to do if somebody tries to set a header that Dropshot really should own (like maybe "date" or "transfer-encoding")?  I can imagine:

- silently ignore it
- let them clobber it
- panic
- make this function return a `Result`, even though it feels like this is something that could be detectable at compile time, except that I don't see how to do that

I hate all of these.  What do you think?

Reply:
I think we let them clobber it. It's either going to let consumers do something ... tricky/sketchy, or be a foot-gun, but it's a foot-gun that would--I think--require careful aiming. In other words, I think it might be a useful escape hatch and is unlikely to cause mysterious errors.